src/engine/src/scheduler/Startup.hpp

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include <entt/entt.hpp>
+
+#include "IScheduler.hpp"
+
+namespace ES::Engine {
+class Registry;
+}
+
+namespace ES::Engine::Scheduler {
+using USystem = std::function<void(Registry &)>;
+
+class Startup : public IScheduler {
+  public:
+    explicit Startup(Registry &registry) : IScheduler(registry) {}
+    void operator()(std::vector<USystem> systems) override
+    {
+        if (!_firstRun)
+        {
+            return;
+        }
+
+        for (auto &system : systems)
+        {
+            system(_registry);
+        }
+
+        _firstRun = false;
+    }
```

Comment:
Rather than running this scheduler each time and checking if it was already consummed, you should remove it from registry.

Reply:
good idea but this implies adding a mechanism to delete them afterwards because we probably don't want to delete stuff from the `_schedulers` map while iterating on it.
I will try to implement this properly

Reply:
It's fine for the moment, but we should take `EmptyRegistry` subject in concideration to call `DeleteScheduler` rather than a callback.

Reply:
You can resolve this conversation but please keep in mind what I said.